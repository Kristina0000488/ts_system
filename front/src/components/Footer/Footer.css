.footer {
  width: 100%;
  justify-content: center;
  background-color: #FFFFFF;
  display: flex;
  flex-direction: row;
  text-align: center;
  color: #B9B9B9;
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  letter-spacing: 0em;
  padding: 20px 0;
}
.extraBlock {
  width: 0px;
}
