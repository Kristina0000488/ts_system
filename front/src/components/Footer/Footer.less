@import '../../styles/varibles.less';
@import '../../styles/mixins.less';


.footer {
    width: 100%;
    justify-content: center;
    background-color: @color_white;
    display: flex;
    flex-direction: row;
    text-align: center;
    color: @color_black_light;
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    letter-spacing: 0em;
    padding: 20px 0;
}

.extraBlock {
    width: 0px;
}