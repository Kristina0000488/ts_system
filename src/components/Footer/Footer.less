@import '../../styles/varibles.less';
@import '../../styles/mixins.less';


.footer {
    width: 100%;
    justify-content: center;
    background-color: @color-white;
    display: flex;
    flex-direction: row;
    text-align: center;
    color: @color-black30;
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    letter-spacing: 0em;
    padding: 20px 0;
}

.extraBlock {
    width: 0px;
}