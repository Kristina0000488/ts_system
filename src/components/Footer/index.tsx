import React, { memo }       from 'react';

import { TypeFooterContent } from '../../types';
import {  contentFooter }    from '../../constants';

import './Footer.css';


interface footerProps {
    content?: TypeFooterContent;   
}

function Footer(props: footerProps) 
{
    const { content=contentFooter } = props;

    return (
        <div className='footer'>
            { content && content.map( ({ text }, id) => 
                <div 
                    key={ id } 
                    className='textFooter'
                >
                    { text }
                </div>
            ) }
        </div>
    );
}

export default memo(Footer);