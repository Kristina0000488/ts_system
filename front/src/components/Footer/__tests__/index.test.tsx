import React    from 'react';
import renderer from 'react-test-renderer';

import Footer   from '../';


it('renders correctly', () => {
  const tree = renderer
    .create(<Footer content={[ { text: 'Hello world' } ]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
