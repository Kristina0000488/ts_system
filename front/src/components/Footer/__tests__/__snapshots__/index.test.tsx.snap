// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="textFooter"
  >
    Hello world
  </div>
</div>
`;
